opm/simulators/flow/KeywordValidation.cpp

```diff
@@ -111,6 +128,17 @@ namespace KeywordValidation
             validateKeywordItems(keyword, m_int_items, errors);
             validateKeywordItems(keyword, m_double_items, errors);
         }
+
+        std::unordered_map<std::string, std::function<void(const DeckKeyword& keyword, std::vector<ValidationError>& errors)>> special_validation {
+            {"BRINE", validateBRINE}
+        };
+
+        const auto& special_it = special_validation.find(keyword.name());
+        if (special_it != special_validation.end()) {
+            const auto& validator = special_it->second;
+            validator(keyword, errors);
+        }
+
```

Comment:
This looks like a good approach to add some very generic tests to keywords. However, maybe it would be good to add/store the map of validation functions as an argument when the object is created. Then actual validation code remains separate from the more general validator object.

Reply:
> However, maybe it would be good to add/store the map of validation functions as an argument when the object is created.

Absolutely - the chosen implementation was just a demonstration which should be as little obtrusive as possible. I'll let it linger for now so others can comment - but if this is generally well received I will address your comment before merging.